ground_strike_feature.py

```diff
@@ -0,0 +1,375 @@
+        - fighter_a_ground_strikes_taken (float): The number of ground strikes taken by fighter a
+        - fighter_b_ground_strikes_hit (float): The number of ground strikes hit by fighter b
+        - fighter_b_ground_strikes_taken (float): The number of ground strikes taken by fighter b
+        - differential (bool): Whether to compute the differential or not
+
+        Returns:
+        - (float, float): The ground strikes percentages for each fighter as a tuple
+        """
+        fighter_a_ground_strikes = fighter_a_ground_strikes_hit / fighter_a_ground_strikes_taken if fighter_a_ground_strikes_taken != 0 else 0
+        fighter_b_ground_strikes = fighter_b_ground_strikes_hit / fighter_b_ground_strikes_taken if fighter_b_ground_strikes_taken != 0 else 0
+
+        if differential:
+            return fighter_a_ground_strikes - fighter_b_ground_strikes, fighter_b_ground_strikes - fighter_a_ground_strikes
+        else:
+            return fighter_a_ground_strikes, fighter_b_ground_strikes
+    
+    def compute_ground_strikes(self, df, fighter_a_id, fighter_b_id, index, round_number, last_fights, column_name, differential=False, is_defense=False, is_percentage=False):
+        """
+        Computes the ground strikes for each fighter in the dataset based on column_name
+
+        Parameters:
+        - df (pd.Dataframe): The original dataframe containing all the fights
+        - fighter_a_id (int): The id of the first fighter
+        - fighter_b_id (int): The id of the second fighter
+        - index (int): The index of the current fight
+        - round_number (int): The round number of the current fight
+        - last_fights (int): The number of last fights to consider
+        - column_name (string): The name of the column to compute
+        - differential (bool): Whether to compute the differential or not
+        - is_defense (bool): Whether to compute the defense or not
+        - is_percentage (bool): Whether to compute the percentage or not
+
+        Returns:
+        - (float, float): The ground strikes for each fighter as a tuple
+        """
+        all_prev_fights = df.loc[:index-1]
+        
+        if not all_prev_fights.empty:
+            fighter_a_id_vals = all_prev_fights.fighter_a_id.values
+            fighter_b_id_vals = all_prev_fights.fighter_b_id.values
+
+            fighter_a_fights = all_prev_fights[(fighter_a_id_vals == fighter_a_id)][-last_fights:]
+            fighter_b_fights = all_prev_fights[(fighter_b_id_vals == fighter_b_id)][-last_fights:]
+
+            sum_a, count_a = self.get_fighter_ground_details(fighter_a_fights, fighter_a_id, round_number, f'total_{column_name}' if round_number == 0 else f'round_{round_number}_{column_name}', is_defense, is_percentage)
+            sum_b, count_b = self.get_fighter_ground_details(fighter_b_fights, fighter_b_id, round_number, f'total_{column_name}' if round_number == 0 else f'round_{round_number}_{column_name}', is_defense, is_percentage)
+
```

Comment:
Unless I am misunderstanding how we're calculating everything we shouldn't be dividing by the number of fights at any point

Reply:
if you just sum it then you just get some huge number but if we average it out for each of the last fights then we get like a real number that makes more sense for the ground strike stats. That is my reasoning. @ivan-guo What do you think?